WooCommerce/Classes/ViewRelated/Blaze/TargetTopics/BlazeTargetTopicPickerViewModel.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+import Yosemite
+import protocol Storage.StorageManagerType
+
+/// View model for `BlazeTargetTopicPickerView`
+final class BlazeTargetTopicPickerViewModel: ObservableObject {
+
+    @Published var selectedTopics: Set<BlazeTargetTopic>?
+    @Published var searchQuery: String = ""
+    @Published private(set) var syncState = SyncState.syncing
+    @Published private var displayedTopics: [BlazeTargetTopic] = []
+    @Published private var fetchedTopics: [BlazeTargetTopic] = []
+    @Published private var isSyncingData: Bool = false
+    @Published private var syncError: Error?
+
+    var shouldDisableSaveButton: Bool {
+        selectedTopics?.isEmpty == true || syncState == .error || syncState == .syncing
+    }
+
+    /// Blaze target device ResultsController.
+    private lazy var resultsController: ResultsController<StorageBlazeTargetTopic> = {
+        let predicate = NSPredicate(format: "locale == %@", locale.identifier)
+        let sortDescriptorByID = NSSortDescriptor(keyPath: \StorageBlazeTargetTopic.name, ascending: true)
+        let resultsController = ResultsController<StorageBlazeTargetTopic>(storageManager: storageManager,
+                                                                           matching: predicate,
+                                                                           sortedBy: [sortDescriptorByID])
+        return resultsController
+    }()
+
+    private let siteID: Int64
+    private let locale: Locale
+    private let stores: StoresManager
+    private let storageManager: StorageManagerType
+    private let onSelection: (Set<BlazeTargetTopic>?) -> Void
+
+    init(siteID: Int64,
+         selectedTopics: Set<BlazeTargetTopic>? = nil,
+         locale: Locale = .current,
+         stores: StoresManager = ServiceLocator.stores,
+         storageManager: StorageManagerType = ServiceLocator.storageManager,
+         onSelection: @escaping (Set<BlazeTargetTopic>?) -> Void) {
+        self.siteID = siteID
+        self.selectedTopics = selectedTopics
+        self.locale = locale
+        self.stores = stores
+        self.storageManager = storageManager
+        self.onSelection = onSelection
+
+        configureResultsController()
+        configureSyncState()
+    }
+
+    @MainActor
+    func syncTopics() async {
+        syncError = nil
+        isSyncingData = true
+        do {
+            try await withCheckedThrowingContinuation { continuation in
+                stores.dispatch(BlazeAction.synchronizeTargetTopics(siteID: siteID, locale: locale.identifier) { result in
+                    switch result {
+                    case .success:
+                        continuation.resume(returning: Void())
+                    case .failure(let error):
+                        continuation.resume(throwing: error)
+                    }
+                })
+            }
+        } catch {
+            DDLogError("⛔️ Error syncing Blaze target devices: \(error)")
```

Comment:
We need to update this error message. 
```suggestion
            DDLogError("⛔️ Error syncing Blaze target topics: \(error)")
```